File: 1813-maximum-erasure-value/1813-maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        res = 0
        inSubarr = set()
        l = 0
        sum = 0
        for num in nums:
            while num in inSubarr:
                sum -= nums[l]
                inSubarr.remove(nums[l])
                l += 1
            inSubarr.add(num)
            sum += num
            res = max(res, sum)
        return res

File: 1813-maximum-erasure-value/1813-maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        int res = 0;
        unordered_set<int> inSubarr;
        int sum = 0;
        int l = 0;
        for(int r = 0; r < nums.size(); r++)
        {
            while(inSubarr.find(nums[r]) != inSubarr.end())
            {
                sum -= nums[l];
                inSubarr.erase(nums[l]);
                l++;
            }
            sum += nums[r];
            inSubarr.insert(nums[r]);
            res = max(res, sum);
        }
        return res;
    }
};